src/thor/costmatrix.cc

```diff
@@ -442,6 +465,8 @@ bool CostMatrix::ExpandInner(baldr::GraphReader& graphreader,
     return true;
   }
 
+  // TODO(nils): refactor this whole thing to only have a single if (FORWARD) {}
```

Comment:
there's a lot of ternaries in this algorithm, most of which can be refactored away. probably the same in most other algorithms.